### ISSUE TITLE
Fix bogus circularity bug

### ISSUE THREAD

**Issue by philbooth** [philbooth/bfj#53]
Continued from https://github.com/philbooth/bfj/issues/47#issuecomment-317730658:

> Moreover I had to set option { circular: 'ignore' } to get it to work, whereas i'm pretty sure there is no circular reference in my dataset.

This is troubling as it suggests there is a bug in the traversal code, which may be producing incorrect results. @rgranger, do you have a dataset you can share that reproduces the problem?

**Comment by rgranger** [comment]
Well, my dataset is quite huge, and i'm afraid that i am not allowed to share it.

But I came up with a minimal dataset that reproduces the bug :

```javascript
const obj = {}

const dataset = {
  a: obj,
  b: obj
}

bfj.stringify(dataset) // => error circular reference
```

Another one :

```javascript
const dataset = {
  a: {},
}
dataset.b = dataset.a

bfj.stringify(dataset) // => error circular reference
```
These two examples are NOT circular, and you can stringify them with JSON.stringify()

Here is a real minimal circular reference :

```javascript
const dataset = {}
dataset.a = dataset
```

**Comment by philbooth** [comment]
Ah yeah, it's just a stupid bug on my part, will fix. Thanks for explaining what a circular reference is, lol.

**Comment by philbooth** [comment]
Should be fixed in `4.1.1`, let me know if you get any problems.
